### ISSUE TITLE
Failed to run symbion example

### ISSUE THREAD

**Issue by yh570** [angr/angr#2858]
---

Hi, I'm very interested in your Symbion tool for concrete+symbolic execution. I'm following https://angr.io/blog/angr_symbion/ and trying to reproduce the Symbion example, but in the 2nd stage, `simgr.run()` didn't find the satisfied path. The error message is shown below:
```
(angr) yh570:~/driller/rex/tests$ python test_symbion.py                                                                         
gdbserver 127.0.0.1:9876 /home/yh570/driller/rex/tests/../binaries/tests/x86_64/packed_elf64
WARNING | 2021-08-11 18:46:07,970 | cle.backends.elf.elf | Segment PT_LOAD is empty at 0x6d1408!
CRITICAL | 2021-08-11 18:46:08,064 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '1'}, 'token': None, 'stream': 'stdout'}
CRITICAL | 2021-08-11 18:46:08,168 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '2'}, 'token': None, 'stream': 'stdout'}
CRITICAL | 2021-08-11 18:46:08,270 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '3'}, 'token': None, 'stream': 'stdout'}
CRITICAL | 2021-08-11 18:46:08,412 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '4'}, 'token': None, 'stream': 'stdout'}
CRITICAL | 2021-08-11 18:46:08,517 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '5'}, 'token': None, 'stream': 'stdout'}
CRITICAL | 2021-08-11 18:46:08,633 | avatar.targets.GDBTarget0.GDBResponseListener | Catched unknown async message: {'type': 'notify', 'message': 'breakpoint-deleted', 'payload': {'id': '6'}, 'token': None, 'stream': 'stdout'}
[1]exploration: <SimulationManager with 1 found>
[2]Symbolically executing binary to find dropping of second stage[ address:  0x400d99 ]
[2]exploration: <SimulationManager with all stashes empty (1 errored)>
Traceback (most recent call last):
  File "test_symbion.py", line 87, in <module>
    new_symbolic_state = exploration.stashes['found'][0]
IndexError: list index out of range
```

The code I'm running is shown below:
```python
import subprocess
import os
import nose
import avatar2 as avatar2
                                                       
import angr
import claripy
from angr_targets import AvatarGDBConcreteTarget


GDB_SERVER_IP = '127.0.0.1'
GDB_SERVER_PORT = 9876

# First set everything up
binary_x64 = os.path.join(os.path.dirname(os.path.realpath(__file__)),
                                          os.path.join('..', 'binaries',
                                          'tests','x86_64',
                                          'packed_elf64'))

# Spawning of the gdbserver analysis environment
print("gdbserver %s:%s %s" % (GDB_SERVER_IP,GDB_SERVER_PORT,binary_x64))
subprocess.Popen("gdbserver %s:%s %s" % (GDB_SERVER_IP,GDB_SERVER_PORT,binary_x64),
                  stdout=subprocess.PIPE,
                  stderr=subprocess.PIPE,
                  shell=True)

# Instantiation of the AvatarGDBConcreteTarget
avatar_gdb = AvatarGDBConcreteTarget(avatar2.archs.x86.X86_64,
                                     GDB_SERVER_IP, GDB_SERVER_PORT)                                                                        

# Creation of the project with the new attributes 'concrete_target'
p = angr.Project(binary_x64, concrete_target=avatar_gdb,
                             use_sim_procedures=True)

entry_state = p.factory.entry_state()
entry_state.options.add(angr.options.SYMBION_SYNC_CLE)
entry_state.options.add(angr.options.SYMBION_KEEP_STUBS_ON_SYNC)

simgr = p.factory.simgr(entry_state)

## Now, let's the binary unpack itself
simgr.use_technique(angr.exploration_techniques.Symbion(find=[0x85b853]))
exploration = simgr.run()
new_concrete_state = exploration.stashes['found'][0]

# Hit the new stub 4 times before having our unpacked code at 0x400cd6
for i in range(0,4):
    simgr = p.factory.simgr(new_concrete_state)
    simgr.use_technique(angr.exploration_techniques.Symbion(find=[0x85b853]))
    exploration = simgr.run()
    new_concrete_state = exploration.stashes['found'][0]

## Reaching the first decision point
simgr = p.factory.simgr(new_concrete_state)
simgr.use_technique(angr.exploration_techniques.Symbion(find=[0x400cd6]))
exploration = simgr.run()
new_concrete_state = exploration.stashes['found'][0]
print("[1]exploration: " + str(exploration))

# Declaring a symbolic buffer
arg0 = claripy.BVS('arg0', 8*32)

# The address of the symbolic buffer would be the one of the
# hardcoded malware configuration
symbolic_buffer_address = new_concrete_state.regs.rbp-0xc0

# Setting the symbolic buffer in memory!
new_concrete_state.memory.store(symbolic_buffer_address, arg0)

simgr = p.factory.simgr(new_concrete_state)

DROP_STAGE2_V2 = 0x400d99
DROP_STAGE2_V1 = 0x400d6a
VENV_DETECTED = 0x400da5
FAKE_CC = 0x400db9

print("[2]Symbolically executing binary to find dropping of second stage" +
       "[ address:  " + hex(DROP_STAGE2_V2) + " ]")

# Symbolically explore the malware to find a specific behavior by avoiding
# evasive behaviors
exploration = simgr.explore(find=DROP_STAGE2_V2, avoid=[DROP_STAGE2_V1,
                                                       VENV_DETECTED, FAKE_CC ])
print("[2]exploration: " + str(exploration))

# Get our synchronized state back!
new_symbolic_state = exploration.stashes['found'][0]


print("[3]Executing binary concretely with solution found until the end " +
hex(BINARY_EXECUTION_END))

simgr = p.factory.simgr(new_symbolic_state)

# Concretizing the solution to reach the interesting behavior in the memory
# of the concrete process and resume until the end of the execution.
simgr.use_technique(angr.exploration_techniques.Symbion(find=[BINARY_EXECUTION_END],
                              memory_concretize = [(symbolic_buffer_address,arg0)],
                              register_concretize=[]))
                              
exploration = simgr.run()
new_concrete_state = exploration.stashes['found'][0]
```

And below is my angr environment report:
```
angr environment report
=============================
Date: 2021-08-11 18:59:04.536861
Running in virtual environment at /home/yh570/.virtualenvs/angr
Platform: linux-x86_64
Python version: 3.7.3 (default, Apr  3 2019, 05:39:12)
[GCC 8.3.0]
######## angr #########
Python found it in /home/yh570/driller/rex/angr-dev/angr/angr
Pip version angr 9.0.gitrolling
Git info:
        Current commit bfba2af1ea2eb941001339f47a1264a685c60eec from branch master
        Checked out from remote origin: https://github.com/angr/angr
######## ailment #########
Python found it in /home/yh570/driller/rex/angr-dev/ailment/ailment
Pip version ailment 9.0.gitrolling
Git info:
        Current commit bef6268dd3d4ea9c251fd24f8a301375771d9dd7 from branch master
        Checked out from remote origin: https://github.com/angr/ailment
######## cle #########
Python found it in /home/yh570/driller/rex/angr-dev/cle/cle
Pip version cle 9.0.gitrolling
Git info:
        Current commit 3909a5ffdb1d4126e0ef359e8013e79350b12a92 from branch master
        Checked out from remote origin: https://github.com/angr/cle
######## pyvex #########
Python found it in /home/yh570/driller/rex/angr-dev/pyvex/pyvex
Pip version pyvex 9.0.gitrolling
Git info:
        Current commit 4a37c8330435f7323036e2bc08f4d6271ed24eae from branch master
        Checked out from remote origin: https://github.com/angr/pyvex
######## claripy #########
Python found it in /home/yh570/driller/rex/angr-dev/claripy/claripy
Pip version claripy 9.0.gitrolling
Git info:
        Current commit 36c640346a822a1950ca43d6d75678e33c731832 from branch master
        Checked out from remote origin: https://github.com/angr/claripy
######## archinfo #########
Python found it in /home/yh570/driller/rex/angr-dev/archinfo/archinfo
Pip version archinfo 9.0.gitrolling
Git info:
        Current commit b150db4c0a939140966df8b0056b6deb5b07efbf from branch master
        Checked out from remote origin: https://github.com/angr/archinfo
######## z3 #########
Python found it in /home/yh570/.virtualenvs/angr/local/lib/python3.7/site-packages/z3
Pip version z3-solver 4.8.5.0
Couldn't find git info
######## unicorn #########
Python found it in /home/yh570/.virtualenvs/angr/local/lib/python3.7/site-packages/unicorn
Pip version unicorn 1.0.2rc4
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/yh570/driller/rex/angr-dev/angr/angr/lib/angr_native.so', handle 276d570 at 0x7f80e9265be0>
unicorn: <CDLL '/home/yh570/.virtualenvs/angr/local/lib/python3.7/site-packages/unicorn/lib/libunicorn.so', handle 1e861e0 at 0x7f80e85af470>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f80e91ac9b0>
z3: <CDLL '/home/yh570/.virtualenvs/angr/lib/python3.7/site-packages/z3/lib/libz3.so', handle 20c3920 at 0x7f80e5f01be0>
```

**Comment by ElNiak** [comment]
I got the same error

**Comment by ElNiak** [comment]
It seems that the "problem" came from angr 9.0. The example works for angr==8.20.7.27 but not after this version. Still investigate.

**Comment by ltfish** [comment]
Without me rerunning the code (I can do that but I don't have cycles at this very moment), can you please post the error/exception in the errored state? You can get it by doing `simgr.errored[0].debug()` or `simgr.errored[0].reraise()`.

**Comment by ElNiak** [comment]
Hello, here is the output for the reraise():
{'type': 'result', 'message': 'error', 'payload': {'msg': 'Unable to read memory.'}, 'token': 590, 'stream': 'stdout'}
Traceback (most recent call last):
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr_targets/targets/avatar_gdb.py", line 40, in read_memory
    res = self.target.read_memory(address, 1, int(nbytes), raw=True)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/avatar2/watchmen.py", line 78, in watchtrigger
    ret = func(self, *args, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/avatar2/targets/target.py", line 35, in check
    return func(self, *args, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/avatar2/targets/target.py", line 386, in read_memory
    return self.protocols.memory.read_memory(address, size, num_words, raw)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/avatar2/protocols/gdb.py", line 686, in read_memory
    raise Exception("Failed to read memory!")
Exception: Failed to read memory!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "SymbionExample.py", line 129, in <module>
    print("simgr.errored[0].reraise() %s", simgr.errored[0].reraise())
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/sim_manager.py", line 818, in reraise
    raise self.error.with_traceback(self.traceback)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/sim_manager.py", line 402, in step_state
    successors = self.successors(state, successor_func=successor_func, **run_args)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/sim_manager.py", line 441, in successors
    return self._project.factory.successors(state, **run_args)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/factory.py", line 60, in successors
    return self.default_engine.process(*args, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/slicing.py", line 19, in process
    return super().process(*args, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/engine.py", line 149, in process
    self.process_successors(self.successors, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/failure.py", line 21, in process_successors
    return super().process_successors(successors, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/syscall.py", line 18, in process_successors
    return super().process_successors(successors, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/hook.py", line 54, in process_successors
    return super().process_successors(successors, procedure=procedure, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/unicorn.py", line 169, in process_successors
    return super().process_successors(successors, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/soot/engine.py", line 65, in process_successors
    return super().process_successors(successors, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/heavy.py", line 138, in process_successors
    self.handle_vex_block(irsb)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/super_fastpath.py", line 24, in handle_vex_block
    super().handle_vex_block(irsb)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/slicing.py", line 26, in handle_vex_block
    super().handle_vex_block(irsb)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/actions.py", line 30, in handle_vex_block
    super().handle_vex_block(irsb)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/inspect.py", line 45, in handle_vex_block
    super().handle_vex_block(irsb)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 446, in handle_vex_block
    self._handle_vex_stmt(stmt)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/slicing.py", line 30, in _handle_vex_stmt
    super()._handle_vex_stmt(stmt)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/inspect.py", line 40, in _handle_vex_stmt
    super()._handle_vex_stmt(stmt)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/resilience.py", line 36, in inner
    return getattr(super(VEXResilienceMixin, self), func)(*iargs, **ikwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/heavy.py", line 204, in _handle_vex_stmt
    super()._handle_vex_stmt(stmt)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 51, in _handle_vex_stmt
    handler(stmt)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 188, in _handle_vex_stmt_WrTmp
    self._analyze_vex_stmt_WrTmp_data(stmt.data)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 184, in _analyze_vex_stmt_WrTmp_data
    def _analyze_vex_stmt_WrTmp_data(self, *a, **kw): return self. _handle_vex_expr(*a, **kw)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/inspect.py", line 31, in _handle_vex_expr
    return super()._handle_vex_expr(expr)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 55, in _handle_vex_expr
    result = handler(expr)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 131, in _handle_vex_expr_GetI
    return self._perform_vex_expr_GetI(
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 142, in _perform_vex_expr_GetI
    return self._perform_vex_expr_GetI_get(offset, ty)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/light/light.py", line 139, in _perform_vex_expr_GetI_get
    def _perform_vex_expr_GetI_get(self, *a, **kw): return self. _perform_vex_expr_Get(*a, **kw)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/actions.py", line 93, in _perform_vex_expr_Get
    result = super()._perform_vex_expr_Get(offset, ty, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/claripy/datalayer.py", line 85, in _perform_vex_expr_Get
    res = super()._perform_vex_expr_Get(offset, ty, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/engines/vex/heavy/heavy.py", line 22, in _perform_vex_expr_Get
    return self.state.registers.load(offset, self._ty_to_bytes(ty), action=action, inspect=inspect)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/unwrapper_mixin.py", line 14, in load
    return super().load(_raw_ast(addr),
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/name_resolution_mixin.py", line 57, in load
    return super().load(addr, size=size, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/bvv_conversion_mixin.py", line 29, in load
    return super().load(addr, size=size, fallback=fallback_bv, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/clouseau_mixin.py", line 81, in load
    r = super().load(addr,
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/actions_mixin.py", line 13, in load
    r = super().load(addr, size=size, condition=condition, fallback=fallback, action=action, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/underconstrained_mixin.py", line 23, in load
    return super().load(addr, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/size_resolution_mixin.py", line 80, in load
    return super().load(addr, size=size, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/size_resolution_mixin.py", line 28, in load
    return super().load(addr, size=out_size, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/address_concretization_mixin.py", line 276, in load
    return self._load_one_addr(self.state.solver.eval(addr), True, addr, condition, size, read_value=None,
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/address_concretization_mixin.py", line 261, in _load_one_addr
    sub_value = super().load(concrete_addr, size=size, condition=sub_condition, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/actions_mixin.py", line 66, in load
    return super().load(addr, action=action, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/conditional_store_mixin.py", line 7, in load
    res = super().load(addr, condition=condition, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/paged_memory/paged_memory_mixin.py", line 129, in load
    vals.append(page.load(pageoff, size=size, endness=endness, page_addr=pageno*self.page_size, memory=self, cooperate=True, **kwargs))
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/paged_memory/pages/ultra_page.py", line 101, in load
    cycle(page_addr + addr + size)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/paged_memory/pages/ultra_page.py", line 61, in cycle
    fill(end)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/paged_memory/pages/ultra_page.py", line 71, in fill
    new_ast = self._default_value(global_start_addr, size,  # pylint: disable=assignment-from-no-return
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/paged_memory/pages/ispo_mixin.py", line 20, in _default_value
    return func(*args, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/underconstrained_mixin.py", line 43, in _default_value
    return super()._default_value(addr, size, name=name, key=key, inspect=inspect, events=events, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/convenient_mappings_mixin.py", line 91, in _default_value
    d = super()._default_value(addr, size, **kwargs)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr/storage/memory_mixins/default_filler_mixin.py", line 15, in _default_value
    mem = self.state.project.concrete_target.read_memory(addr, size)
  File "/home/crochetch/Documents/toolchain_malware_analysis/penv/lib/python3.8/site-packages/angr_targets/targets/avatar_gdb.py", line 44, in read_memory
    raise SimConcreteMemoryError("AvatarGDBConcreteTarget can't read_memory at address %x exception %s" % (address, e))
angr.errors.SimConcreteMemoryError: AvatarGDBConcreteTarget can't read_memory at address 388 exception Failed to read memory!

**Comment by jomanchu-twosix** [comment]
I also ran into this issue, with the same error as reported by @ElNiak. After debugging this a little, I wonder if it is related to #2957?

I started a pdb shell for this state (`sim.errored[0].debug()`) and went up the call stack many times until I reached `/home/angr/angr-dev/angr/angr/engines/vex/heavy/inspect.py(45)handle_vex_block()`.

```
ipdb> import pyvex
ipdb> for s in irsb.statements:
    if isinstance(s, pyvex.stmt.IMark):
        print(s)

------ IMark(0x400cd6, 1, 0) ------
------ IMark(0x400cd7, 2, 0) ------
------ IMark(0x400cd9, 0, 0) ------
```

I believe this implies that the instruction boundaries were 0x400cd6, 0x400cd7, 0x400cd9. Opening `packed_elf64` in Ghidra (i.e., the binary is still packed) and disassembling at 0x400cd6, I see
```
0x400cd6 XCHG EAX, EDX
0x400cd7 FUCOM
0x400cd9 OUTSB DX, RSI
```

I believe this implies that angr is seeing the state of memory that it loaded at the beginning of the analysis run, not the state of memory after the Symbion exploration.

**Comment by AlexCasual** [comment]
What news, there are any suggestions to fix it?

**Comment by rhelmot** [comment]
the status on symbion is that nobody on the core team contributed any code to symbion and we don't really have the cycles to learn. symbion was built by a handful of people outside the core team, notably @degrigis, who aren't as available to respond to issues. I'm sorry if this is a disappointing answer, but it's the state of things right now.

**Comment by degrigis** [comment]
Hey there! So, many things changed in `angr` since I wrote the Symbion use cases, I would definitely not rely on that code anymore. Please refer to [this testcase](https://github.com/angr/angr-targets/blob/master/tests/test_concrete_not_packed_elf64.py) as a working example of the technique. 
We recently got a [PR](https://github.com/angr/angr/pull/3327) that fixed #2957, but please be aware of [this issue](https://github.com/angr/angr/issues/3346).
Finally, for the current state of the work related to Symbion, refer to this [issue/RFC](https://github.com/angr/angr/issues/2701).

As @rhelmot said, unfortunately I don't have many cycles to maintain Symbion nowadays, but I'll try to orchestrate the conversations around it and keep at least the use-case referred above working.
